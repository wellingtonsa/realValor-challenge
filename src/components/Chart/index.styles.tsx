import styled from 'styled-components';

export const Container = styled.div`
    position: relative;
    background: #FFF;
    padding: 40px;
    width: 505px;
    border-radius: 4px;
    animation: show 1s;
    animation-fill-mode: forwards;
    
    @keyframes show {
        0%{
            opacity: 0;
        }
        50%{
            opacity: 0.5;
        }
        100%{
            opacity: 1;
        }
    }
`;

export const Details = styled.div<{show:boolean}>`
    position: relative;
    background: #334495;
    color: #fff;
    padding-top: 10px;
    width: 505px;
    animation: show 1s;
    animation-fill-mode: forwards;
    cursor: pointer;
    .header {
        width: 100%;
        padding: 10px;
        border: 1px solid #fff;
        border-radius: 4px 4px 0px 0px;
    }

    .content {
        border: 1px solid #fff;
        border-top: 0px;
        background: #fff;
        color: #3c3c3c;
        opacity: ${props => props.show?'1': '0'};
        transition: 500ms;
        padding: 10px;

    }
    
    @keyframes show {
        0%{
            opacity: 0;
        }
        50%{
            opacity: 0.5;
        }
        100%{
            opacity: 1;
        }
    }
`