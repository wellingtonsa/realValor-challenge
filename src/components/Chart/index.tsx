import React, { useState } from 'react';
import { Line } from 'react-chartjs-2';
import moment from 'moment';
import { Container, Details } from './index.styles';

interface IDataSet {
    label: string;
    backgroundColor: string;
    data:  number[]
};

export interface IChart {
    labels: string[];
    datasets: IDataSet[];
};

interface IProps {
    data: IChart
};

type Props = IProps; 

const Chart: React.SFC<Props> = ({ data }) => {
    moment.locale('pt-br');

    const [showDetails, setShowDetails] = useState<boolean>(false)
    
    const details = {
        startDate: data.labels[0],
        startValue: data.datasets[0].data[0],
        endValue: data.datasets[0].data[data.datasets[0].data.length - 1],
        amount: data.datasets[0].data[data.datasets[0].data.length - 1] - data.datasets[0].data[0],
    };  
    console.log(details)
    const handleShowDetails = () => {
        setShowDetails(!showDetails);
    }
    return (
        <>
        <Container>
            <h3>{`Veja aqui os resultados para ${data.datasets[0].label}`}</h3>
            <Line
            options={{ responsive: true }}
            data={data}/>
        </Container>
        <Details show={showDetails}>
            <div className="header" onClick={handleShowDetails}>
                <span>Clique aqui para ver detalhes</span>
            </div>
            <div className="content">
            <div className="line">
                <span>Data de início:</span>  
                <span>{details.startDate}</span>
            </div>
            <div className="line">
                <span>Valor inicial:</span>  
                <span>{details.startValue.toFixed(2)}</span>
            </div>
            <div className="line">
                <span>Valor final:</span>  
                <span>{details.endValue.toFixed(2)}</span>
            </div>
            <div className="line">
                <span>Lucro/Prejuízo:</span>  
                <span style={{ color: (details.amount > 0?'#00F300':'#F30000')}}>{details.amount.toFixed(2)}</span>
            </div>
            </div>
        </Details>
        </>
    )
}

export default Chart;