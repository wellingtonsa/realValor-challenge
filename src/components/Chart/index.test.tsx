import React from "react";
import { shallow } from "enzyme";
import Chart from "./index";


const props = {
  data: {
    labels: [
      'Label 1'
    ],
    datasets:[
    {
        label: 'Cotação 01',
        backgroundColor: '#FFF',
        data: [1]
    }
    ]
  }
};

describe("Chart component", () => {
  const component = shallow(<Chart {...props} />);

  it("should render chart component", () => {
    expect(component).toMatchSnapshot();
  });


  it("check if props render correctly", () => {
    expect(component.find('h3').text()).toEqual(`Veja aqui os resultados para ${props.data.datasets[0].label}`);
  });

});