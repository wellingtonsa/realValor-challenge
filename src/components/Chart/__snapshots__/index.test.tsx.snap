// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart component should render chart component 1`] = `
<Fragment>
  <styled.div>
    <h3>
      Veja aqui os resultados para Cotação 01
    </h3>
    <Line
      data={
        Object {
          "datasets": Array [
            Object {
              "backgroundColor": "#FFF",
              "data": Array [
                1,
              ],
              "label": "Cotação 01",
            },
          ],
          "labels": Array [
            "Label 1",
          ],
        }
      }
      options={
        Object {
          "responsive": true,
        }
      }
    />
  </styled.div>
  <styled.div
    show={false}
  >
    <div
      className="header"
      onClick={[Function]}
    >
      <span>
        Clique aqui para ver detalhes
      </span>
    </div>
    <div
      className="content"
    >
      <div
        className="line"
      >
        <span>
          Data de início:
        </span>
        <span>
          Label 1
        </span>
      </div>
      <div
        className="line"
      >
        <span>
          Valor inicial:
        </span>
        <span>
          1.00
        </span>
      </div>
      <div
        className="line"
      >
        <span>
          Valor final:
        </span>
        <span>
          1.00
        </span>
      </div>
      <div
        className="line"
      >
        <span>
          Lucro/Prejuízo:
        </span>
        <span
          style={
            Object {
              "color": "#F30000",
            }
          }
        >
          0.00
        </span>
      </div>
    </div>
  </styled.div>
</Fragment>
`;
